src/float.rs

```diff
@@ -0,0 +1,85 @@
+use core::ops::*;
+use core::num::One;
+
+/// Trait for some basic operations on floats
+pub trait Float<T: Sized + Copy + Clone + Sub<Output=T> + ShlAssign<T> + Shl<T, Output=T> + One> {
+    /// Returns the bitwidth of the float type
+    fn bits() -> T;
+
+    /// Returns the bitwidth of the significand
+    fn significand_bits() -> T;
+
+    /// Returns the leading zeroes of `a`
+    fn rep_clz(a: T) -> T;
+
+    /// Returns a normalized version of the significand
+    fn normalize(significand: &mut T) -> T {
+        let shift = Self::rep_clz(*significand) - Self::rep_clz(T::one() << Self::significand_bits());
+        *significand <<= shift as T;
+        T::one() - shift as T
+    }  
+}
+
+impl Float<u32> for f32 {
+    fn bits() -> u32 {
+        32
+    }
+    fn significand_bits() -> u32 {
+        23
+    }
+    fn rep_clz(a: u32) -> u32 {
+        #[cfg(not(msvc))]
+        extern "C" {
+            fn __builtin_clz(a: u32) -> u32;
```

Comment:
Where does this symbol come from? And is this even necessary? Couldn't this just use `u32::leading_zeros`? That method (and the u64 variant) already lower to an LLVM intrinsic: `llvm.ctlz.i*` in LLVM IR.